gap/oper.gi

```diff
@@ -2043,6 +2043,59 @@ function(D, root)
   return visited;
 end);
 
+InstallMethod(VerticesReachableFrom, "for a digraph and a list of vertices",
+[IsDigraph, IsList],
+function(D, roots)
+    local accessible_from_node, candidate, root, i, visited, visited_as_ints, N;
+    N := DigraphNrVertices(D);
+    visited := BlistList([1 .. N], []);
+    for root in roots do
+      if not(visited[root]) then
+        accessible_from_node := VerticesReachableFrom(D, root);
+        for candidate in accessible_from_node do
+          if not(visited[candidate]) then
+            visited[candidate] := true;
+          fi;
+        od;
+      fi;
+    od;
+
+    visited_as_ints := [];
+
+    for i in [1 .. N] do
+      if visited[i] then;
+        Add(visited_as_ints, i);
+      fi;
+    od;
+
+    return visited_as_ints;
+end);
+
+InstallMethod(IsOrderIdeal, "for a digraph and a list of vertices",
+[IsDigraph, IsList],
+# Check if digraph represents a partial order
+function(D, roots)
+  local reachable_vertices, vertex_in_subset, i, N;
+  if not(IsPartialOrderDigraph(D)) then
+    ErrorNoReturn("the 1st element (a digraph) must be a partial order digraph");
```

Comment:
Please add a test that covers these lines, you can just a the gap input to the test file:
~~~
gap> IsOrderIdeal(D, [1, 2, 3]); # where D is not a partial order 
~~~
and then do 
~~~
Test("tst/standard/oper.tst", rec(rewriteToFile := true));
~~~
and GAP will add the expected output to the test file for you.

Reply:
Neat - thanks!